File: jobboardmobile/lib/dto/application_dto.dart
import 'company_dto.dart';
import 'job_dto.dart';

class ApplicationDTO {
  final int id;
  final String employeeName;
  final int userId;
  final JobDTO jobDTO;
  final CompanyDTO companyDTO;
  final bool approved; // Added field

  ApplicationDTO({
    required this.id,
    required this.employeeName,
    required this.userId,
    required this.jobDTO,
    required this.companyDTO,
    required this.approved, // Updated constructor
  });

  factory ApplicationDTO.fromJson(Map<String, dynamic> json) {
    return ApplicationDTO(
      id: json['id'] ?? 0,
      employeeName: json['employeeName'] ?? '',
      userId: json['userId'] ?? 0,
      jobDTO: JobDTO.fromJson(json['jobDTO'] ?? {}),
      companyDTO: CompanyDTO.fromJson(json['companyDTO'] ?? {}),
      approved: json['approved'] ?? false, // Updated fromJson
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'employeeName': employeeName,
      'userId': userId,
      'jobDTO': jobDTO.toJson(),
      'companyDTO': companyDTO.toJson(),
      'approved': approved, // Updated toJson
    };
  }
}

File: JobBoardService/src/main/java/com/project4/JobBoardService/DTO/ApplicationDTO.java
package com.project4.JobBoardService.DTO;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class ApplicationDTO {
    private Long id;
    private String employeeName;
    private Long userId;
    private JobDTO jobDTO;
    private CompanyDTO companyDTO;
    private byte[] cvFile;
    private String coverLetter;
    private boolean approved; // Added field
    private boolean isNew; // Added field
    private LocalDateTime createdAt; // Added field
}
